lib/rate-limit.js

```diff
@@ -1,13 +1,17 @@
 const attempts = {}
 
 export default function rateLimit (instance, maxRetry = 5) {
+  function handleError (error) {
+    instance.defaults.errorEmitter.emit('error', error)
```

Comment:
I have to say I agree with @Khaledgarbaya here. Two types of events ( `log` and `error` ) feels cleaner. :)